Dockerfile

```diff
@@ -121,21 +120,28 @@ RUN \
     && unzip ./SCHEMAS_OPENGIS_NET.zip "ogcapi/*" -d /schemas.opengis.net \
     && rm -f ./SCHEMAS_OPENGIS_NET.zip \
 
-    # Install remaining pygeoapi deps
-    && pip3 install -r requirements-docker.txt \
-
-    # Install pygeoapi
-    && pip3 install -e . \
-
-    # Set default config and entrypoint for Docker Image
-    && cp /pygeoapi/docker/default.config.yml /pygeoapi/local.config.yml \
-    && cp /pygeoapi/docker/entrypoint.sh /entrypoint.sh  \
-
     # Cleanup TODO: remove unused Locales and TZs
+    # NOTE: this tries to remove gcc, but the actual package gcc-11 can't be
+    #       removed because python3-scipy depends on python3-pythran which
+    #       depends on g++
     && apt-get remove --purge -y gcc ${DEB_BUILD_DEPS} \
     && apt-get clean \
     && apt autoremove -y  \
     && rm -rf /var/lib/apt/lists/*
 
+ADD requirements-docker.txt /pygeoapi
```

Comment:
You can avoid this line, by moving line 136 here

Reply:
Thanks for the review!

Yes, this would be possible. However then any change in the pygeoapi source would lead to a reinstall of the pip packages.

The use case this optimizes for is when some development happens and the code is changed while `requirements-docker.txt` stays the same. Then a rebuild of the docker image is extremely fast.

Reply:
@totycro @doublebyte1 shall we have a multi stage build and two dockerfiles, one for development and one for production?

Reply:
My preference would be to stick to 1 Dockerfile as i've argued here: https://github.com/geopython/pygeoapi/pull/1252#issuecomment-1602187908

Having 2 Dockerfiles would also work, but with some more maintainence overhead. I'm not sure how multi-stage build could be used in a convenient way here.

Reply:
Agree for a single Dockerfile.